tests/benchmark/s2n_negotiate_client_benchmark.cc

```diff
@@ -0,0 +1,279 @@
+        if (DEBUG_PRINT) {
+            printf("Error in negotiate!\n");
+        }
+    }
+
+    if (DEBUG_PRINT) {
+        printf("Connected to %s:%s\n", host, port);
+    }
+
+    GUARD_EXIT(s2n_connection_free_handshake(conn), "Error freeing handshake memory after negotiation");
+
+    s2n_blocked_status blocked;
+    int shutdown_rc = s2n_shutdown(conn, &blocked);
+    if (shutdown_rc == -1 && blocked != S2N_BLOCKED_ON_READ) {
+        fprintf(stderr, "Unexpected error during shutdown: '%s'\n", s2n_strerror(s2n_errno, "NULL"));
+        exit(1);
+    }
+
+    GUARD_EXIT(s2n_connection_free(conn), "Error freeing connection");
+
+    GUARD_EXIT(s2n_config_free(config), "Error freeing configuration");
+
+    free(unsafe_verify_data);
+}
+
+static void ClientBenchmark(benchmark::State& state) {
+    int i;
+    for(i = 0; i < WARMUP_ITERS; i++) {
+        client_benchmark(state, true);
+    }
+    for (auto _ : state) {
+        state.PauseTiming();
+        client_benchmark(state, false);
+    }
+    state.SetBytesProcessed(state.iterations() * sizeof(int));
+}
+
+int Client::start_benchmark_client(int argc, char** argv) {
+    s2n_init();
+    char file_prefix[100];
+    char bench_format[100] = "--benchmark_out_format=";
+
+    while (1) {
+        int c = getopt(argc, argv, "c:i:w:o:t:sD");
+        if (c == -1) {
+            break;
+        }
+
+        switch (c) {
+            case 0:
+                /* getopt_long() returns 0 if an option.flag is non-null (Eg "parallelize") */
+                break;
+            case 'c':
+                use_corked_io = atoi(optarg);
+                break;
+            case 'i':
+                ITERATIONS = atoi(optarg);
+                break;
+            case 'w':
+                WARMUP_ITERS = atoi(optarg);
+                break;
+            case 'o':
+                strcpy(file_prefix, optarg);
+                break;
+            case 't':
+                strcat(bench_format, optarg);
+                break;
+            case 's':
+                insecure = 1;
+                break;
+            case 'D':
+                DEBUG_PRINT = 1;
+                break;
+            case '?':
+            default:
+                fprintf(stdout, "getopt returned: %d", c);
+                break;
+        }
+    }
+
+    if (optind < argc) {
+        host = argv[optind++];
+    }
+
+    if (optind < argc) {
+        port = argv[optind++];
+    }
+
+    char **newv = (char**)malloc((argc + 4) * sizeof(*newv));
+    memmove(newv, argv, sizeof(*newv) * argc);
+    char bench_out[100] = "--benchmark_out=client_";
+    strcat(bench_out, file_prefix);
+    newv[argc] = bench_out;
+    char aggregate[100] = "--benchmark_display_aggregates_only=true";
+    newv[argc+1] = aggregate;
+    newv[argc+2] = bench_format;
+    newv[argc+3] = 0;
+    argc+=3;
+    argv = newv;
+
+    setup_config();
+    unsigned int len = sizeof(all_suites) / sizeof(all_suites[0]);
```

Comment:
```suggestion
    unsigned int num_cipher_suites = sizeof(all_suites) / sizeof(all_suites[0]);
```
`len` is a bad variable name and makes the for loop on 262 hard to understand.